### ISSUE TITLE
RpbErrorResp errcode should be set more proper value

### ISSUE THREAD

**Issue by taichi** [basho/riak_kv#336]
Once error occurs on PBC API, RpbErrorResp is returned but it doesn't make sense because RIAKC_ERR_GENERAL is enable. It should be set the same value to REST API.

It doesn't matter for client if server returns not found on GET operation. So heavy performance cost exception should not be triggered. If modified is returned, application should choose to trigger to repair data or just to retry get ops. It's not server-side issue.

All human readable fields e.g. errmsg field should be understandable for user. The contents should be tuned in long development cycle, so instability in early release is acceptable. If we requires detailed implementation on error handling, it should be done by reading errcode field.
I suggest we improve current version of PBC API.

**Comment by seancribbs** [comment]
@taichi This is already on the roadmap, but won't make it until the September release. Although I agree that the error code should be more meaningful, making it the same as the HTTP error code would be misleading; instead we can provide meaningful, Riak-specific error codes. Thank you for your input.

**Comment by evanmcc** [comment]
Did this make it in at some point?

**Comment by seancribbs** [comment]
@evanmcc No, we currently still return `0` for all err codes. This issue needs a champion.
